### ISSUE TITLE
show_error and show_debug_message demanding more arguments.

### ISSUE THREAD

**Issue by SuperRiderTH** [enigma-dev/enigma-dev#1055]
```Too few arguments to function `show_debug_message': provided 1, required 2.```
```Too few arguments to function `show_error': provided 2, required 4.```

**Comment by JoshDreamland** [comment]
I believe you're affected by the same JDI bug others have been posting, recently. Have you tried [this workaround​](http://enigma-dev.org/forums/index.php?topic=2669.msg25839#msg25839)?

**Comment by SuperRiderTH** [comment]
That workaround appears to work? I am getting a segfault when trying to run my large project, but that appears to be a separate issue.

**Comment by SuperRiderTH** [comment]
Or maybe not? It is complaining about this:
```
Request for type of self: returning int!
New level: 0
code from here: e=2.0;};
Writing events
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fe29c6bc430, pid=11165, tid=0x00007fe29d86c700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_131-b11) (build 1.8.0_131-b11)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.131-b11 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libstdc++.so.6+0x121430]  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::compare(char const*) const+0x10
#

```

**Comment by JoshDreamland** [comment]
I don't believe I've seen a segfault thrown from there, before. See if you can narrow a smaller test case. The good news is that it doesn't seem to be JDI causing the problem.

**Comment by SuperRiderTH** [comment]
I'm attempting to narrow where the segfault is occurring, and it appears to be happening in "compile_writeDefraggedEvents." So I added some debug output to the different sections, and it appears to be occurring somewhere in the for loop at line 105, ```for (evfit it = used_events.begin(); it != used_events.end(); it++)```

**Comment by JoshDreamland** [comment]
Thanks for tracking that down! My guess is that someone is calling `erase()` without updating the iterator, first. I'll have a look this weekend. (I'm out of state right now and not in a good position to look at code, much less update it.)

**Comment by faissaloo** [comment]
Just came across this, the string bug is causing JDI to fill in all the parameters of a function with NULLs

**Comment by JoshDreamland** [comment]
This persists after 7b767491ddb86ba3b2ab9f0d859cbc160a771bdc?

**Comment by faissaloo** [comment]
Yes, I was working on a fresh clone of the repo

**Comment by JoshDreamland** [comment]
But the workaround still corrects it? We may want to commit dummy headers and augment JDI's search path to prefer them....

**Comment by faissaloo** [comment]
Yes, the workaround still corrects it

**Comment by RobertBColton** [comment]
As you can see, someone reported this again #1102.

**Comment by JoshDreamland** [comment]
@faissaloo Could you make sure #1106 corrects this for you, as well?

**Comment by faissaloo** [comment]
@JoshDreamland Yes, that fixes it

**Comment by fundies** [comment]
Confirmed as fixed in #1106 by other users. I'm going to close this but if you have issue still please reopen it
